desktop/api/src/main/java/org/datacleaner/windows/AbstractWindow.java

```diff
@@ -49,18 +49,22 @@ public AbstractWindow(WindowContext windowContext) {
         setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
         addWindowListener(this);
         getContentPane().setBackground(WidgetUtils.BG_COLOR_BRIGHT);
+
     }
 
+    protected abstract boolean isWindowMaximized();
```

Comment:
and naming wise I think it should say "isInitiallyMaximized" because otherwise it gives the impression that it will always tell you the state of the window (which may change, but the "return false" implementation never changes).